launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java

```diff
@@ -250,6 +250,34 @@ public void testMissingAppResource() {
     new SparkSubmitCommandBuilder().buildSparkSubmitArgs();
   }
 
+  @Test
+  public void testIsClientMode() {
+    // Default master is "local[*]"
+    SparkSubmitCommandBuilder launcher =
+            newCommandBuilder(Collections.emptyList());
+    assertTrue("By default application run in local mode",
+            launcher.isClientMode(Collections.EMPTY_MAP));
+    // --master yarn or it can be any RM
+    List<String> sparkSubmitArgs = Arrays.asList(
+            parser.MASTER,
+            "yarn");
+    launcher = newCommandBuilder(sparkSubmitArgs);
+    assertTrue("By default deploy mode is client",
+            launcher.isClientMode(Collections.EMPTY_MAP));
+    // --master yarn and set spark.submit.deployMode to client
+    Map<String, String> userProps = new HashMap<>();
+    userProps.put("spark.submit.deployMode", "client");
+    assertTrue(launcher.isClientMode(userProps));
+    // --master mesos --deploy-mode cluster
+    sparkSubmitArgs = Arrays.asList(
+            parser.MASTER,
+            "mesos",
+            parser.DEPLOY_MODE,
+            "cluster");
+    launcher = newCommandBuilder(sparkSubmitArgs);
+    assertTrue(!launcher.isClientMode(Collections.EMPTY_MAP));
```

Comment:
`assertTrue(!` -> `assertFalse(`.

Reply:
Finally, use `Collections.emptyMap()` instead of `Collections.EMPTY_MAP`. `Collections.EMPTY_MAP` gives us `Unchecked assignment` warning due to type mismatch.